posthog/session_recordings/session_recording_helpers.py

```diff
@@ -149,6 +149,8 @@ def is_unchunked_snapshot(event: Dict) -> bool:
         is_snapshot = event["event"] == "$snapshot"
     except KeyError:
         raise ValueError('All events must have the event name field "event"!')
+    except TypeError:
```

Comment:
So in this case we end up with event being a string of length one?

Could we validate this earlier as well? It seems odd that it is the check for if this is a snapshot that we check this, maybe closer to the bit that's pulling out `batch` can validate?

Also fine to go along as is as you're simply following the existing pattern.

Reply:
No, event is a string of arbitrary length. The problem is not the length, the problem is the type: it's a string instead of a dictionary.

Reply:
Ah yeah, I see what you mean by length one, that's the case here, but other malformed payloads could also raise the same error.

I do agree that we could validate this earlier though. There is an `assert` in line 258 that could be expanded a bit:

```python
if isinstance(data, dict):
    if data.get("batch"):  # posthog-python and posthog-ruby
        data = data["batch"]
        
        if data is None or not isinstance(data, list) or not isinstance(data, dict):
            return cors_response(
                request,
                generate_exception_response(
                    "capture",
                    f"Invalid payload: Payload is empty or not parsable", code="invalid_payload"
                ),
            )
```

Reply:
However, I don't know fully know the implications of dropping that assert for a response, maybe a caller is doing something with an `AssertionError`.

Reply:
I'm also not a super fan of so many `isinstance` calls... Maybe something to think a bit more for later?

Reply:
👍  yup possibly a rabbit hole, fine to get in as is and can iterate later if you want.